euth/projects/mixins.py

```diff
@@ -0,0 +1,23 @@
+from django.core import exceptions
+
+from euth.phases import models as phases_models
+from . import models
+
+class ProjectMixin():
+
+    def dispatch(self, *args, **kwargs):
+        if not 'project' in kwargs:
+            msg = '{} requires `project` view kwarg'.format(self.__cls__.__name__)
+            raise exceptions.ImproperlyConfigured(msg)
+
+        self.project = models.Project.objects.get(slug=kwargs['project'])
+        self.phase = phases_models.Phase.objects.active_phase(self.project)
+        self.module = self.phase.module if self.phase else None
+        return super(ProjectMixin, self).dispatch(*args, **kwargs)
+
+    def get_context_data(self, *args, **kwargs):
+        context = super(ProjectMixin, self).get_context_data(*args, **kwargs)
+        context['project'] = self.project
+        context['phase'] = self.phase
+        context['module'] = self.module
```

Comment:
You can simply use `view.project` or `view.phase` in a template, because `django.views.generic.base.ContextMixin` adds the current view to the context. This makes adding class attributes to the context superfluous.

If you are not sure if all views will use `django.views.generic.base.ContextMixin` you can simply extend it with your mixin.